extensions/cornerstone-3d/src/initMeasurementService.js

```diff
@@ -212,17 +212,19 @@ const connectMeasurementServiceToTools = (
         return;
       }
 
-      const { id, label } = measurement;
+      const { uid, label } = measurement;
 
-      const sourceAnnotation = annotation.state.getAnnotation(id);
-
-      if (sourceAnnotation) {
-        sourceAnnotation.label = label;
-        if (sourceAnnotation.hasOwnProperty('text')) {
+      const sourceAnnotation = annotation.state.getAnnotation(uid);
+      const data = sourceAnnotation.data;
+      if (data) {
+        data.label = label;
+        if (data.hasOwnProperty('text')) {
           // Deal with the weird case of ArrowAnnotate.
```

Comment:
Maybe we should really just make the arrow annotate tool change the label of its annotation rather than add a proprietary `text` field. We seem to have to do this over and over.